src/profile/components/additionalInformation/__tests__/AdditionalInformation.test.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+
+import {
+  cloneProfileAndProvideManipulationFunctions,
+  getMyProfile,
+} from '../../../../common/test/myProfileMocking';
+import {
+  renderComponentWithMocksAndContexts,
+  TestTools,
+  cleanComponentMocks,
+} from '../../../../common/test/testingLibraryTools';
+import { Language, ProfileData } from '../../../../graphql/typings';
+import AdditionalInformation from '../AdditionalInformation';
+import {
+  MockedResponse,
+  ResponseProvider,
+} from '../../../../common/test/MockApolloClientProvider';
+import {
+  additionalInformationType,
+  AdditionalInformationValue,
+} from '../../../helpers/editData';
+import i18n from '../../../../common/test/testi18nInit';
+import RenderChildrenWhenDataIsComplete from '../../../../common/test/RenderChildrenWhenDataIsComplete';
+import profileConstants from '../../../constants/profileConstants';
+
+describe('<AdditionalInformation /> ', () => {
+  const responses: MockedResponse[] = [];
+  const initialProfile = getMyProfile().myProfile as ProfileData;
+  const renderTestSuite = () => {
+    const responseProvider: ResponseProvider = () =>
+      responses.shift() as MockedResponse;
+    return renderComponentWithMocksAndContexts(
+      responseProvider,
+      <RenderChildrenWhenDataIsComplete>
+        <AdditionalInformation />
+      </RenderChildrenWhenDataIsComplete>
+    );
+  };
+  const t = i18n.getFixedT('fi');
+  beforeEach(() => {
+    responses.length = 0;
+  });
+  afterEach(() => {
+    cleanComponentMocks();
+  });
+
+  const initTests = async (): Promise<TestTools> => {
+    responses.push({ profileData: initialProfile });
+    const testTools = await renderTestSuite();
+    await testTools.fetch();
+    return Promise.resolve(testTools);
+  };
+
+  it('should render language selection, auto save and show success/error message', async () => {
```

Comment:
This one test has quite a many responsibilities. It seems that it could be split at least into two, the success and error cases.